data/components/main-toolbar.js

```diff
@@ -88,7 +89,8 @@ var MainToolbar = React.createClass({
         ),
         ToolbarButton({bsSize: "xsmall", onClick: this.onSwitchPerspective},
           perspectiveLabel
-        )
+        ),
+        ConnectionFilter(this.props)
```

Comment:
Yes, I agree. It would be better if the dropdown appeared to the left of the search box.

I assume SearchBox is in `componentDidMount` because it is a legacy non-react component, and needs to be injected directly into the DOM through a constructor method.

Because they (SearchBox and ConnectionFilter) are both `float: right` and css is weird, I need the ability to position SearchBox above ConnectionFilter in the markup. This would mean being able to put SearchBox in the render method, and thus rewriting SearchBox to react. I'll have a stab at it and see what I can do.